src/Reflection/ReflectionParameter.php

```diff
@@ -388,12 +401,17 @@ public function getPosition() : int
      */
     public function getTypeHint() : ?Type
     {
+        return $this->getRealClassName($this->node->type);
+    }
+
+    private function getRealClassName($className) : ?Type
```

Comment:
It is enough to add a type-hint into doc comment? It can be a string, NULL, or PhpParser's node.